swarm/src/connection.rs

```diff
@@ -239,12 +239,17 @@ where
                     ));
                     continue;
                 }
-                Poll::Ready(Some((info, Err(error)))) => {
-                    handler.on_connection_event(ConnectionEvent::ListenUpgradeError(
-                        ListenUpgradeError { info, error },
-                    ));
-                    continue;
-                }
+                Poll::Ready(Some((info, Err(error)))) => match error {
+                    ConnectionHandlerUpgrErr::Upgrade(error) => {
+                        handler.on_connection_event(ConnectionEvent::ListenUpgradeError(
+                            ListenUpgradeError { info, error },
+                        ));
+                        continue;
+                    }
+                    ConnectionHandlerUpgrErr::Timeout => {
+                        log::debug!("Timeout expired during an inbound substream negotiation")
```

Comment:
```suggestion
                        log::debug!("Failed to negotiate protocol on inbound stream within {seconds}s")
```

That would be more useful IMO.

Reply:
`seconds` doesn't seem to exist in this scope, we probably can make `ConnectionHandlerUpgrErr::Timeout` have it . I will implement after clarifying https://github.com/libp2p/rust-libp2p/pull/3307#discussion_r1064264754

Reply:
> `seconds` doesn't seem to exist in this scope, we probably can make `ConnectionHandlerUpgrErr::Timeout` have it . I will implement after clarifying https://github.com/libp2p/rust-libp2p/pull/3307#discussion_r1064264754

Yes, it was only meant to be illustrative on what I'd like to have printed!